pusher/pusher_client.py

```diff
@@ -69,6 +78,13 @@ def trigger(self, channels, event_name, data, socket_id=None):
 
         data = data_to_string(data, self._json_encoder)
 
+        if is_encrypted_channel(channels[0]):
+            # set a random seed
+            random.seed(datetime.now())
+            # generate a 24-bytes random nonce as it is required by the encrypt() function (needed by nacl)
+            nonce = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(24))
```

Comment:
The choice of using an external random generator and not the one provided by pynacl is driven by the fact that in order to test certain parts of the crypto module the nonce should be externally specifiable and should allow deterministic results. Thats why I went with the python's std lib `random` generator, which should be sufficient for the nonce requirements and allows fixed seeding for reproducibility.

As you cleverly suggested we could take the nonce as an optional parameter i.e. `nonce=None` so that it allows optional external deterministic nonce passing and default internal cryptographically secure pseudo-random nonce generation.